File: backend/src/main/java/com/example/backend/global/exception/GlobalException.java
package com.example.backend.global.exception;

import org.springframework.http.HttpStatus;

/**
 * GlobalException
 * 전역에서 발생할 수 있는 커스텀 예외 처리 클래스
 * @author 100minha
 */
public class GlobalException extends RuntimeException {
	private final GlobalErrorCode globalErrorCode;

	public GlobalException(GlobalErrorCode globalErrorCode) {
		super(globalErrorCode.getMessage());
		this.globalErrorCode = globalErrorCode;
	}

	public HttpStatus getStatus() {
		return globalErrorCode.getHttpStatus();
	}

	public String getCode() {
		return globalErrorCode.getCode();
	}
}

File: backend/src/main/java/com/example/backend/global/exception/GlobalException.kt
package com.example.backend.global.exception

import org.springframework.http.HttpStatus

/**
 * GlobalException
 * 전역에서 발생할 수 있는 커스텀 예외 처리 클래스
 * @author 100minha
 */
class GlobalException(private val globalErrorCode: GlobalErrorCode) :
    RuntimeException() {

    override val message: String
        get() = globalErrorCode.message

    val status: HttpStatus
        get() = globalErrorCode.httpStatus

    val code: String
        get() = globalErrorCode.code
}
